consensus/spos/subround.go

```diff
@@ -130,6 +130,7 @@ func (sr *Subround) DoWork(rounder consensus.Rounder) bool {
 			}
 		case <-time.After(rounder.RemainingTime(startTime, maxTime)):
 			if sr.Extend != nil {
+				sr.RoundCanceled = true
```

Comment:
why cancelling the round ? this is on extend, not ?

Reply:
When the method Extend is called, means that time is out and round should be canceled wherever the computation is. There could be an edge case in the new model, as the received header would be asynchronous during the current round, and if the extend method is called the revert account state is done and the commit of the possible late received header should not be done as it could generate problem if it is  done in the same time with the call of RevertAccounState method from extend. Now is safe as if RoundCanceled is true this action would be avoided